builtin/update-index.c

```diff
@@ -1098,7 +1099,13 @@ int cmd_update_index(int argc, const char **argv, const char *prefix)
 	argc = parse_options_end(&ctx);
 
 	getline_fn = nul_term_line ? strbuf_getline_nul : strbuf_getline_lf;
+	if (mark_skip_worktree_only && gvfs_config_is_set(-1))
+		die("Modifying the skip worktree bit is not supported on a GVFS repo");
+	
 	if (preferred_index_format) {
+		if (preferred_index_format != 4 && gvfs_config_is_set(-1))
+			die("Changing the index version is not supported on a GVFS repo");
```

Comment:
Oh, and should we talk about a `GVFS repo`, or rather about a `VFS for Git repository`?

Reply:
We haven't renamed any of the GVFS code/flags/messages yet so I left this the same so the search/replace that happens at some point in the future can catch this one as well.